python/taichi/main.py

```diff
@@ -151,17 +151,62 @@ def example(self, arguments: list = sys.argv[2:]):
             help=f"Name of an example (supports .py extension too)\n",
             type=TaichiMain._example_choices_type(choices),
             choices=sorted(choices))
+        parser.add_argument(
+            '-p',
+            '--print',
+            required=False,
+            dest='print',
+            action='store_true',
+            help="Print example source code instead of running it")
+        parser.add_argument(
+            '-P',
+            '--pretty-print',
+            required=False,
+            dest='pretty_print',
+            action='store_true',
+            help="Like --print, but print in a rich format with line numbers")
+        parser.add_argument(
+            '-s',
+            '--save',
+            required=False,
+            dest='save',
+            action='store_true',
+            help="Save source code to current directory instead of run")
         args = parser.parse_args(arguments)
 
         examples_dir = TaichiMain._get_examples_dir()
         target = str((examples_dir / f"{args.name}.py").resolve())
         # path for examples needs to be modified for implicit relative imports
         sys.path.append(str(examples_dir.resolve()))
-        print(f"Running example {args.name} ...")
 
         # Short circuit for testing
         if self.test_mode: return args
 
+        if args.save:
+            print(f"Saving example {args.name} to current directory...")
+            shutil.copy(target, '.')
+            return
+
+        if args.pretty_print:
+            try:
+                import rich.syntax
+                import rich.console
+            except ImportError as e:
+                print('To make -P work, please: python3 -m pip install rich')
+                raise e
```

Comment:
Instead of re-raising the exception, how about return 1?

Reply:
Yes, good idea!
```suggestion
                return 1
```